lib/stat_tracker.rb

```diff
@@ -57,13 +57,82 @@ def percentage_visitor_wins
     (away_win?.to_f / @games.size).round(2)
   end
 
+  def percentage_ties
+    tie_games = @game_teams.find_all do |game|
+      game.game_result.include?("TIE")
+    end
+    results = tie_games.length.to_f / @game_teams.length.to_f
+    results.round(2)
+  end
+
+  def count_of_games_by_season
+    ids = []
+    only_ids = []
+      @games.find_all do |game|
+        ids = game.season_id
+        only_ids << ids
+      end
+    uniq_ids = only_ids.uniq
+    count_games = []
+    final_count = Hash.new
+      uniq_ids.each do |id|
+        count_games = @games.find_all {|game|game.season_id.include?(id)}
+        final_count[id] = count_games.length
+      end
+      final_count
+  end
+
   # LEAGUE STATS
 
   def count_of_teams
     @teams.count
   end
 
+  def highest_scoring_home_team
+    game_team_goals = Hash.new
+    @game_teams.each do |game_team|
+      game_team_goals[game_team.team_id.to_sym] = []
+    end
+    @game_teams.each do |game_team|
+      goal_array = game_team_goals[game_team.team_id.to_sym]
+        if game_team.home_or_away == "home"
+          goal_array << game_team.goals.to_i
+        end
+    end
+    game_team_goals.map do |team, goal_array|
+      game_team_goals[team] = (goal_array.sum.to_f / goal_array.length.to_f)
+    end
+    max_key_value = game_team_goals.max_by{ |team, goals| goals }
+  
+    team = @teams.find do |team|
+      max_key_value[0].to_s == team.team_id
+    end
+    team.team_name
+  end
+  
+  def lowest_scoring_home_team
+    game_team_goals = Hash.new
+    @game_teams.each do |game_team|
+      game_team_goals[game_team.team_id.to_sym] = []
+    end
+    @game_teams.each do |game_team|
+      goal_array = game_team_goals[game_team.team_id.to_sym]
+        if game_team.home_or_away == "home"
+          goal_array << game_team.goals.to_i
+        end
+    end
+    game_team_goals.map do |team, goal_array|
+      game_team_goals[team] = (goal_array.sum.to_f / goal_array.length.to_f)
+    end
+    min_key_value = game_team_goals.min_by{ |team, goals| goals }
+    team = @teams.find do |team|
```

Comment:
If we end up having time, these methods would be nice to refactor given that there's so much shared logic with the best/worst offense methods.